packages/react-components/react-tabster/src/focus/focusWithinPolyfill.ts

```diff
@@ -0,0 +1,61 @@
+import { KEYBORG_FOCUSIN, KeyborgFocusInEvent, createKeyborg, disposeKeyborg } from 'keyborg';
+import { FOCUS_WITHIN_CLASS } from './constants';
+
+/**
+ * Because `addEventListener` type override falls back to 2nd definition (evt name is unknown string literal)
+ * evt is being typed as a base class of MouseEvent -> `Event`.
+ * This type is used to override `listener` calls to make TS happy
+ */
+type ListenerOverride = (evt: Event) => void;
+
+export function applyFocusWithinPolyfill(element: HTMLElement, win: Window): () => void {
+  const keyborg = createKeyborg(win);
+
+  // When navigation mode changes to mouse, remove the focus-within selector
+  keyborg.subscribe(isNavigatingWithKeyboard => {
+    if (!isNavigatingWithKeyboard) {
+      removeFocusWithinClass(element);
+    }
+  });
+
+  // Keyborg's focusin event is delegated so it's only registered once on the window
+  // and contains metadata about the focus event
+  const keyborgListener = (e: KeyborgFocusInEvent) => {
+    if (keyborg.isNavigatingWithKeyboard() && isHTMLElement(e.target)) {
+      // Griffel can't create chained global styles so use the parent element for now
+      applyFocusWithinClass(element);
+    }
+  };
+
+  // Make sure that when focus leaves the scope, the focus within class is removed
+  const blurListener = (e: FocusEvent) => {
+    if (!e.relatedTarget || (isHTMLElement(e.relatedTarget) && !element.contains(e.relatedTarget))) {
+      removeFocusWithinClass(element);
+    }
+  };
+
+  element.addEventListener(KEYBORG_FOCUSIN, keyborgListener as ListenerOverride);
+  element.addEventListener('focusout', blurListener);
+
+  // Return disposer
+  return () => {
+    element.removeEventListener(KEYBORG_FOCUSIN, keyborgListener as ListenerOverride);
+    element.removeEventListener('focusout', blurListener);
+    disposeKeyborg(keyborg);
+  };
+}
+
+function applyFocusWithinClass(el: HTMLElement) {
+  el.classList.add(FOCUS_WITHIN_CLASS);
+}
+
+function removeFocusWithinClass(el: HTMLElement) {
+  el.classList.remove(FOCUS_WITHIN_CLASS);
+}
+
+function isHTMLElement(target: EventTarget | null): target is HTMLElement {
+  if (!target) {
+    return false;
+  }
+  return Boolean(target && typeof target === 'object' && 'hasAttribute' in target && 'getAttribute' in target);
```

Comment:
nit: this check looks strange (and unsafe?) as `hasAttribute` & `getAttribute` are not used in this file

Reply:
So remember how sometimes `intanceof HTMLElement` does not work ? This is because globals are different between frames and windows: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof#instanceof_and_multiple_contexts_e.g._frames_or_windows

We should actually stop using `instanceof` because intrinsically it's dangerous for multi-window

Reply:
Then we need to ban `instanceof` with ESLint, would you mind to create an issue for that?

However my comment had a different context: _"You're checking for some methods, but don't use them. You use `.classlist`, but don't do checks for it"_

Reply:
I've updated the check to use `classList`  and `contains` which are used